yarn-project/bb-prover/src/avm_proving_tests/avm_proven_token.test.ts

```diff
@@ -1,119 +1,40 @@
-
-    await tester.simProveVerify(
-      /*sender=*/ admin,
-      /*setupCalls=*/ [],
-      /*appCalls=*/ [
-        {
-          fnName: 'constructor',
-          args: constructorArgs,
-          address: token.address,
-        },
-      ],
-      /*teardownCall=*/ undefined,
-      /*expectRevert=*/ false,
-    );
+  beforeAll(async () => {
+    // FULL PROVING! Not check-circuit.
+    tester = await AvmProvingTester.new(/*checkCircuitOnly=*/ false, /*globals=*/ defaultGlobals(), metrics);
   });
 
-  it('token mint, transfer, burn', async () => {
-    const mintAmount = 100n;
-    const transferAmount = 50n;
-    const authwitNonce = new Fr(0);
-
-    await checkBalance(sender, 0n);
-
-    await tester.simProveVerify(
-      /*sender=*/ admin,
-      /*setupCalls=*/ [],
-      /*appCalls=*/ [
-        {
-          fnName: 'mint_to_public',
-          args: [/*to=*/ sender, mintAmount],
-          address: token.address,
-        },
-      ],
-      /*teardownCall=*/ undefined,
-      /*expectRevert=*/ false,
-    );
-    await checkBalance(sender, mintAmount);
-
-    await tester.simProveVerify(
-      /*sender=*/ sender,
-      /*setupCalls=*/ [],
-      /*appCalls=*/ [
-        {
-          fnName: 'transfer_in_public',
-          args: [/*from=*/ sender, /*to=*/ receiver, transferAmount, authwitNonce],
-          address: token.address,
-        },
-      ],
-      /*teardownCall=*/ undefined,
-      /*expectRevert=*/ false,
-    );
-    await checkBalance(sender, mintAmount - transferAmount);
-    await checkBalance(receiver, transferAmount);
-
-    await tester.simProveVerify(
-      /*sender=*/ receiver,
-      /*setupCalls=*/ [],
-      /*appCalls=*/ [
-        {
-          fnName: 'burn_public',
-          args: [/*from=*/ receiver, transferAmount, authwitNonce],
-          address: token.address,
-        },
-      ],
-      /*teardownCall=*/ undefined,
-      /*expectRevert=*/ false,
-    );
-    await checkBalance(receiver, 0n);
+  afterAll(() => {
+    if (process.env.BENCH_OUTPUT) {
+      mkdirSync(path.dirname(process.env.BENCH_OUTPUT), { recursive: true });
+      writeFileSync(process.env.BENCH_OUTPUT, metrics.toGithubActionBenchmarkJSON());
+    } else if (process.env.BENCH_OUTPUT_MD) {
+      writeFileSync(process.env.BENCH_OUTPUT_MD, metrics.toPrettyString());
+    } else {
+      logger.info(`\n`); // sometimes jest tests obscure the last line(s)
+      logger.info(metrics.toPrettyString());
+    }
   });
 
-  const checkBalance = async (account: AztecAddress, expectedBalance: bigint) => {
-    const balResult = await tester.simulateTx(
-      sender,
-      /*setupCalls=*/ [],
-      /*appCalls=*/ [
-        {
-          address: token.address,
-          fnName: 'balance_of_public',
-          args: [/*owner=*/ account],
-          isStaticCall: true,
-        },
-      ],
-    );
-    expect(balResult.revertCode.isOK()).toBe(true);
-    expectAppCall0Output(balResult, [new Fr(expectedBalance)]);
-  };
+  it(
```

Comment:
I haven't checked that my bench changes work well with many txs. Could you confirm?